### PR TITLE
Add support for multiple CouchDB databases, multiple namespaces, and `readOnly` configurations

### LINKED ISSUE #7359: Support more than one Couch database for a single Couch Server
**Is your feature request related to a problem? Please describe.**
Open MCT should be able to support more than one couch database on the same server.

**Describe the solution you'd like**
A developer should be able to install the couch plugin and optionally pass in multiple database names (eg. `openmct-sandbox`, `openmct-cm`) and have Open MCT populate the tree with a separate root object and namespace for each database.

The Couch plugin already supports [multiple namespaces](https://github.com/nasa/openmct/blob/master/src/plugins/persistence/couch/plugin.js#L45), so this should be a relatively trivial change. I think the solution is to implement a new instance of the CouchObjectProvider per namespace.

Also see https://github.com/nasa/openmct/issues/7360 where we implement support for "read-only" databases.

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7359 #7360

### Describe your changes:

Add support for multiple CouchDB databases, namespaces, and `readOnly` configurations.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/persistence/couch/README.md
Comment (ozyx) [inline src/plugins/persistence/couch/README.md:183]:
```suggestion
- `additionalNamespaces`: Other namespaces that this plugin should respond to requests for.
```
backticks here?

**Review by ozyx — APPROVED** [review]
Wow, this is awesome. I haven't tried it, but I assume this works with multiple CouchDB databases as well?

I did find one issue with search, where after the search has completed, the "loading" progress bar is still spinning permanently:


https://github.com/nasa/openmct/assets/9259993/06931601-3c5e-4062-809d-8af979ace969

I think we can spin that off into its own issue, though. I don't have any PR-blocking comments at this time! I tested locally and everything seems to work well.

**Comment by ozyx** [comment]
Found another issue @scottbell . I don't think it should block the PR at this point? It's still possible to add annotations to a domain object in a read-only database. Is that expected?

**Comment by scottbell** [comment]
> I did find one issue with search, where after the search has completed, the "loading" progress bar is still spinning permanently:

@ozyx Thank you for the review! I wasn't able to reproduce the spinning search thing. Was there anything in the console?

https://github.com/nasa/openmct/assets/9853862/867c7231-18af-4a26-91f8-8ebfd1f0e957

**Comment by scottbell** [comment]
> Found another issue @scottbell . I don't think it should block the PR at this point? It's still possible to add annotations to a domain object in a read-only database. Is that expected?

@ozyx I think that's fine as long as the `namespaceToSaveAnnotations` property is set to a writable database. I just tested and threw an error `Uncaught (in promise) Object provider does not support saving`, which seems reasonable to me. Note a variation of this was already happening with YAMCS telemetry, which doesn't support saving.

**Comment by unlikelyzero** [comment]
@scottbell while you're waiting for additional review from Andrew, can you make sure our backup and restore scripts are up to date with this pattern?

**Comment by scottbell** [comment]
> @scottbell while you're waiting for additional review from Andrew, can you make sure our backup and restore scripts are up to date with this pattern?

@unlikelyzero good idea! I’ll take a look

**Comment by scottbell** [comment]
> > @scottbell while you're waiting for additional review from Andrew, can you make sure our backup and restore scripts are up to date with this pattern?
> 
> @unlikelyzero good idea! I’ll take a look

@unlikelyzero they seem like they still work, provided you're using the default db name of `openmct`, but that's always been the case.

File: src/plugins/persistence/couch/CouchObjectProvider.js
Comment (akhenry) [inline src/plugins/persistence/couch/CouchObjectProvider.js:345]: Is this a private function?

File: src/plugins/persistence/couch/CouchObjectProvider.js
Comment (akhenry) [inline src/plugins/persistence/couch/CouchObjectProvider.js:369]: Is this a private function?

File: src/plugins/persistence/couch/plugin.js
Comment (akhenry) [inline src/plugins/persistence/couch/plugin.js:50]: chef's kiss 🤌

File: src/plugins/persistence/couch/plugin.js
Comment (akhenry) [inline src/plugins/persistence/couch/plugin.js:62]: Love it

**Review by akhenry — CHANGES_REQUESTED** [review]
This looks great, thank you. My only request is that for consistency with the default namespace we remove the logic to create the root if it doesn't exist. Users can install My Items for additional namespaces if they desire, and the `omitRoot` logic might be confusing.

File: src/plugins/persistence/couch/CouchObjectProvider.js
Comment (scottbell) [reply src/plugins/persistence/couch/CouchObjectProvider.js:345]: it should be!

File: src/plugins/persistence/couch/CouchObjectProvider.js
Comment (scottbell) [reply src/plugins/persistence/couch/CouchObjectProvider.js:369]: it should be!

File: .cspell.json
Comment (akhenry) [inline .cspell.json:496]: I'm prepared to accept unnormalized at a stretch. But unnnormalized? With three n's?

**Review by akhenry — APPROVED** [review]

**Pull request merged**
